pkg/utils/utils_test.go

```diff
@@ -0,0 +1,84 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+func TestToKubernetesResourceName(t *testing.T) {
```

Comment:
Most (if not all) of the tests are hardly realistic scenarios because K8s just will not allow non - RFC 1123 DNS subdomain name as a CM or Secret name (which we use to construct Volume name) ?

Reply:
I've simplified the unit test cases there to focus on what's makes sense to me.